src/riak_kv_get_core.erl

```diff
@@ -284,6 +288,36 @@ num_pr(GetCore = #getcore{num_pok=NumPOK, idx_type=IdxType}, Idx) ->
             GetCore
     end.
 
+%% @private Print a warning if objects are not equal. Only called on case of no read-repair
+%% This situation could happen with pre 2.1 vclocks in very rare cases. Fixing the object
+%% requires the user to rewrite the object in 2.1+ of Riak. Logic is disabled when using
+%% a defined version of aae hashing as it's no longer needed.
+maybe_log_old_vclock(Results) ->
+    case riak_core_capability:get({riak_kv, object_hash_version}, undefined) of
```

Comment:
So right now, as I understand it, we don't start the hashtree upgrades as soon as the capability flips, because we want to make sure all partitions have exchanged first so that we get a chance to log any inconsistent data here in this function. However, it looks like we don't do any logging here once the capability flips, right? I might be missing something but it seems like we may want to be checking something other than the capability to decide whether to check for mismatched data.